### PR TITLE
Multiple Validation Fixes for DummyClassifiers

### PR DESCRIPTION
**Pull request by esherman-credo**
## Describe your changes
Removed second passage of `message` in logger warning message.
Remove requirement that `data.X` be supplied for Performance and Fairness evaluators
Update `model.type` assignment for DummyClassifier and Keras classifiers.

## Issue ticket number and link
Various slack messages...

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] I have built basic tests for new functionality (particularly new evaluators)
- [ ] If new libraries have been added, I have checked that [readthedocs](https://readthedocs.org/projects/credoai-lens/) API documentation is constructed correctly
- [ ] Will this be part of a major product update? If yes, please write one phrase about this update.

## Extra-mile Checklist
- [ ] I have thought expansively about edge cases and written tests for them

### CODE REVIEW

File: credoai/artifacts/model/classification_model.py [hunk 1]
```diff
@@ -117,11 +123,16 @@ def __post_init__(self):
...
         elif self.model_info["framework"] == "credoai":
             # Functionality for DummyClassifier
-            self.model_like = getattr(self.model_like, "model_like", None)
+            if self.model_like.model_like is not None:
+                self.model_like = self.model_like.model_like
             # If the dummy model has a model_like specified, reassign
             # the classifier's model_like attribute to match the dummy's
             # so that downstream evaluators (ModelProfiler) can use it
 
+            self.type = self.model_like.type
```
Comment (IanAtCredo) [inline credoai/artifacts/model/classification_model.py:132]: Do we need to do it like this? Can't we just count the number of unique values that are created by the predict function?

Or perhaps we can do this to automatically determine the type within DummyClassifier...

File: credoai/artifacts/model/classification_model.py [hunk 2]
```diff
@@ -141,6 +152,13 @@ class DummyClassifier:
     model_like : model_like, optional
         While predictions are pre-computed, the model object, itself, may be of use for
         some evaluations (e.g. ModelProfiler).
+    binary_clf : bool, optional, default = True
```
Comment (IanAtCredo) [inline credoai/artifacts/model/classification_model.py:155]: maybe an "auto" option would be useful? When "auto", classification will be determined by the `predict_output` vector

**Review by IanAtCredo — APPROVED** [review]
Non-blocking questions/suggestions for alternative implementations.

**Review by amrasekh — APPROVED** [review]

File: credoai/artifacts/model/classification_model.py (hunk 1 above)
Comment (esherman-credo) [reply credoai/artifacts/model/classification_model.py:132]: That occurred to me but it's possible the supplied predictions don't actually contain samples with labels for all possible classes and thus we might incorrectly infer that it's supposed to be a binary problem when it's really multi-class.

File: credoai/artifacts/model/classification_model.py (hunk 2 above)
Comment (esherman-credo) [reply credoai/artifacts/model/classification_model.py:155]: See above comment. Imagine `predict_output` is [0, 0, 0...] because the event is rare and the model can't tease out when something is likely to be class 1. Auto would treat this as...binary? What if there are two rare events (class 1 and 2)? We'd miss that.
